app/database/itemStore.go

```diff
@@ -38,9 +38,31 @@ func (s *ItemStore) Insert(data *Item) error {
 	return s.db.insert(s.tableName(), data)
 }
 
+// All creates a composable query without filtering
+func (s *ItemStore) All() *DB {
+	return &DB{s.db.Table(s.tableName())}
+}
+
 // IsValidHierarchy gets an ordered set of item ids and returns whether they forms a valid item hierarchy path from a root
 func (s *ItemStore) IsValidHierarchy(ids []int64) (bool, error) {
-	return false, nil
+	l := len(ids) - 1
+	if l < 1 {
+		return false, fmt.Errorf("not enough ids %v", ids)
```

Comment:
meaning `len(ids) < 2` ? there is no problem of having a hierarchy of 1 element... i.e., here asking for the breadcrumb for the root only should be valid

Reply:
side remark: When you do testing (and especially unit testing and you add a case such as this one, it MUST be included in your tests!